app/models/custom_value.rb

```diff
@@ -79,16 +97,9 @@ def validate_format_of_value
 
   def validate_type_of_value
     if value.present?
-      # Format specific validations
-      case custom_field.field_format
-      when 'int'
-        errors.add(:value, :not_a_number) unless value =~ /\A[+-]?\d+\z/
-      when 'float'
-        begin; Kernel.Float(value); rescue; errors.add(:value, :invalid) end
-      when 'date'
-        errors.add(:value, :not_a_date) unless value =~ /\A\d{4}-\d{2}-\d{2}\z/
-      when 'list'
-        errors.add(:value, :inclusion) unless custom_field.possible_values.include?(value)
+      result = strategy.validate_type_of_value
+      if result
+        errors.add(:value, result)
```

Comment:
I liked keeping the errors interface out of the strategy... Indeed I thought about returning a list, but it was never neccessary I so I thought I'd keep it simple and if someone needs it, he will build it.